Content.Server/Pinpointer/PinpointerSystem.cs

```diff
@@ -210,4 +275,80 @@ private Distance CalculateDistance(Vector2 vec, PinpointerComponent pinpointer)
         else
             return Distance.Far;
     }
+
+    /// <summary>
+    ///     Clears the list with stored targets and turns off the pinpointer.
+    /// </summary>
+    private void DeleteStoredTargets(EntityUid uid, PinpointerComponent component, EntityUid? user)
+    {
+        component.StoredTargets.Clear();
+        if (component.IsActive)
+        {
+            TogglePinpointer(uid, component);
+        }
+
+    }
+
+    /// <summary>
+    ///     Adds a verb that allows the user to search for the closest target containing a certain component, a verb that
+    ///     allows the user to select any of the stored targets and a verb that allows the user to clear the stored targets.
+    /// </summary>
+    private void OnPinpointerVerb(EntityUid uid, PinpointerComponent component, GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanInteract || args.Hands == null)
+            return;
+
+        //Adds the closest target verb if there is at least 1 stored component.
+        if (component.Components.Count > 0)
+        {
+            foreach (var targetComponent in component.Components)
+            {
+                args.Verbs.Add(new Verb()
+                {
+                    Text = Loc.GetString(component.ComponentNames[component.Components.IndexOf(targetComponent)]),
+                    Act = () => LocateTarget(uid, component, args.User, targetComponent),
```

Comment:
Having 2 disparate fields that correlate like this is a really bad idea, especially when there's no validation being done on the yaml, and please never use IndexOf as it will re-scan the entire list for no reason. Just use the componentregistry.

Reply:
Should I make an if statement for every component instead of the list?